pkg/middleware/middleware_test.go

```diff
@@ -265,287 +267,192 @@ func TestMiddlewareContext(t *testing.T) {
 			})
 		})
 
-		middlewareScenario("When auth_proxy is enabled enabled and user exists", func(sc *scenarioContext) {
```

Comment:
I haven't run the code so maybe I'm missing something here, but shouldn't this test still work after the changes? I see further tests that are related to auth_proxy without ldap, it seems like a bad idea to remove them. Let's sync up.

Reply:
Me and @xlson had talk about it - the tests are changed a little: description, tree-like structure report and default values hoisted from test blocks, plus added couple more, that's the gist of it :).

It seems we are good here